src/main/java/utils/MyDate.java

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright (c) 2021 Fundacion Jala.
+ * This software is the confidential and proprietary information of Fundacion Jala
+ * ("Confidential Information"). You shall not disclose such Confidential
+ * Information and shall use it only in accordance with the terms of the
+ * license agreement you entered into with Fundacion Jala
+ */
+
+package utils;
+
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class MyDate {
+
+    private static final int POSITIVE_VALUE = 1;
+    private static final int NEGATIVE_VALUE = -1;
+    private static final int MATCHER_GROUP_1 = 1;
+    private static final int MATCHER_GROUP_2 = 2;
+    private static final int MATCHER_GROUP_3 = 3;
+    private static final String DATE_SIMPLE_FORMAT = "MM/dd/uuuu";
+    private int value;
+    private String timeUnit;
+    private String time;
+    private Calendar c;
+    private Date parsedDate;
+
+    /**
+     * Gets a date from a valid input String. Valid inputs are:
+     * Strings -> "TODAY", "TOMORROW", "YESTERDAY"
+     * String with format-> {A number}
+     *      {"seconds", "minutes", "hours", "days", "weeks", "months", "years"}
+     *      {"ago", "from now"}
+     * E.g.: "3 hours ago", "1 day from now", "100 minutes from now"
+     * String representing a date with format MM/DD/YYYY, year greater or equal to 1900.
+     * E.g.: "01/12/2000", "12/31/1990"
+     *
+     * @param input
+     * @return Date
+     * @throws Exception
+     */
+    public Date get(final String input) throws Exception {
+        if (validateInput(input)) {
+            return calculateDate();
+        } else {
+            if (isSimpleDate(input)) {
+                return parsedDate;
+            } else {
+                throw new Exception("The input doesn't match a valid entry");
+            }
+        }
+    }
+
+    /**
+     * Verify if input date has valid format.
+     *
+     * @param simpleDate
+     * @return boolean
+     */
+    private boolean isSimpleDate(final String simpleDate) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(DATE_SIMPLE_FORMAT);
+        try {
+            LocalDate localDate = LocalDate.parse(simpleDate, formatter);
+            ZoneId defaultZoneId = ZoneId.systemDefault();
+            this.parsedDate = Date.from(localDate.atStartOfDay(defaultZoneId).toInstant());
+            return true;
+        } catch (Exception e) {

```

Comment:
You should do something with the exception

Reply:
Returning an exception in a higher method.